server/app/routes/v1/ping_api.rb

```diff
@@ -1,13 +1,20 @@
 module V1
   class PingApi < Roda
+    include RequestHelpers
 
     plugin :json
     plugin :render, engine: 'json.jbuilder', views: 'app/views/v1/ping'
 
     route do |r|
       r.is do
         r.get do
-          render('show')
+          begin
+            Grid.count # test db connection
+            MongoPubsub.actor.alive? # test pubsub
+            render('show')
+          rescue
+            halt_request(500, {error: 'Internal server error'})
```

Comment:
Maybe also log the exception for diagnostic purposes? Then it would show up in the logs if the healthcheck fails and the service gets restarted.

Not entirely sure how to include `Kontena::Logging` for `error exc` in the roda routes, though.

Reply:
`rescue => exc` and something to also pass it to `Logger#error` so that you get the error class + message + backtrace